src/w2grid.js

```diff
@@ -2092,11 +2092,13 @@
                 var record = this.get(changes[c].recid);
                 for (var s in changes[c]) {
                     if (s == 'recid') continue; // do not allow to change recid
-                    var key = "." + s;
-                    if (w2utils.isFloat(s)) key = "['" + s + "']";
-                    if (String(s).indexOf("'") != -1) key = "['" + s.replace(/'/g, "\\'") + "']";
+                    if(typeof changes[c][s] === "object") changes[c][s] = changes[c][s].text;
```

Comment:
need space after if and before (. See https://github.com/vitmalina/w2ui/wiki/Contribution-Requirement